### PR TITLE
Jkmarx/fb select all feature

### LINKED ISSUE #3268: No easy way to add lots of files to a visualisation tool
* Version of the web browser and OS: Chrome, Windows 10

### Steps to reproduce
Please list all the actions and the input data used:

1. Open dataset you can run tools on
2. Open tool panel
3. Pick any tool

### Observed behavior
* Can only add one file at a time

### Desired behavior
* Ability to add files quicker, either a button to select all files in dataset, or fewer steps to add a file e.g. check boxes instead of the arrow so it's one click instead of two.

### PR DESCRIPTION
**Pull request by jkmarx**
Closes #3268 

select-all-enhancement

### CODE REVIEW

File: refinery/data_set_manager/utils.py [hunk 1]
```diff
@@ -734,14 +734,16 @@ def format_solr_response(solr_response):
...
-    if solr_response_json.get('facets'):
+    if solr_response_json.get('facets') and include_facet_count != 'false':
         solr_response_json['facet_field_counts'] = create_facet_field_counts(
             solr_response_json.get('facets')
         )
-        del solr_response_json['facets']
     else:
         solr_response_json['facet_field_counts'] = {}
 
+    if solr_response_json.get('facets'):
```
Comment (ilan-gold) [inline refinery/data_set_manager/utils.py]: did you have a preference for leaving this up here as opposed to down with the other chunk of clean-up code at 744?

Comment (jkmarx) [reply refinery/data_set_manager/utils.py]: I'm okay with it being here, but if you feel strongly about it I can relocate it.

Comment (ilan-gold) [reply refinery/data_set_manager/utils.py]: all good with me, just checking

File: refinery/ui/source/js/file-browser/ctrls/select-all-button-ctrl.js [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+      var assayFiles = fileBrowserFactory.assayFiles;
+      var totalAssayCount = fileBrowserFactory.assayFilesTotalItems.count;
+      var selectionGroup = getSelectionGroup();
+
+      // current assay files is larger than 100 need to use API for full list
+      if (totalAssayCount > maxFileCount) {
+        vm.updatingSelectionStatus = true;
+        var deselectFlag = vm.isAllSelected;
+        var assayFilesQuery = assayFileService.query(getCurrentParams());
+        assayFilesQuery.$promise.then(function (response) {
```
Comment (ilan-gold) [inline refinery/ui/source/js/file-browser/ctrls/select-all-button-ctrl.js]: can we add error handling here? thanks!

**Pull request merged**